### ISSUE TITLE
Gemma fails with large genotype matrices

### ISSUE THREAD

**Issue by pjotrp** [genetics-statistics/GEMMA#183]
#182 reports an integer overflow in blas. The stack trace reads

```
GEMMA 0.98 (2018-09-28) by Xiang Zhou and team (C) 2012-2018
Reading Files ... 
## number of total individuals = 876
## number of analyzed individuals = 876
## number of covariates = 1
## number of phenotypes = 1
## number of total SNPs/var        =  1000000
## number of analyzed SNPs         =  1000000
Start Eigen-Decomposition...
pve estimate =0.969141
se(pve) =0.0263771
ERROR: Enforce failed for N>0 in src/fastblas.cpp at line 197 in fast_cblas_dgemm

Thread 1 "gemma-0.98-linu" received signal SIGINT, Interrupt.
0x0000000001571caf in raise ()
(gdb) bt
#0  0x0000000001571caf in raise ()
#1  0x000000000045dea4 in __enforce_fail (
    __function=0x162c3c0 <fast_cblas_dgemm(char const*, char const*, double, gsl_matrix const*, gsl_matrix const*, double, gsl_matrix*)::__func__> "fast_cblas_dgemm", __line=197, __file=0x162c2d4 "src/fastblas.cpp", __assertion=0x162c2e5 "N>0") at src/debug.h:150
#2  fast_cblas_dgemm (TransA=TransA@entry=0x1721e97 "T", TransB=TransB@entry=0x164e863 "N", alpha=alpha@entry=1, A=0x1a8b020, 
    B=B@entry=0x7fffffffc3c0, beta=beta@entry=0, C=C@entry=0x7fffffffc3f0) at src/fastblas.cpp:197
#3  0x000000000045f0ac in fast_dgemm (TransA=TransA@entry=0x1721e97 "T", TransB=TransB@entry=0x164e863 "N", alpha=alpha@entry=1, 
    A=<optimized out>, B=B@entry=0x7fffffffc3c0, beta=beta@entry=0, C=C@entry=0x7fffffffc3f0) at src/fastblas.cpp:222
#4  0x000000000047cbaf in LMM::<lambda(size_t)>::operator()(size_t) const (__closure=__closure@entry=0x7fffffffc5f0, l=0, 
    l@entry=<error reading variable: That operation is not available on integers of more than 8 bytes.>) at src/lmm.cpp:1506
#5  0x000000000047d99f in LMM::Analyze(std::function<std::tuple<std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >, std::vector<double, std::allocator<double> > > (unsigned long)>&, gsl_matrix const*, gsl_vector const*, gsl_matrix const*, gsl_vector const*, gsl_matrix const*, gsl_vector const*, std::set<std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >, std::less<std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > >, std::allocator<std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > > >) (this=this@entry=0x7fffffffcca0, fetch_snp=..., U=<optimized out>, U@entry=0x1a8b020, 
    eval=<optimized out>, eval@entry=0x1a8afb0, UtW=<optimized out>, UtW@entry=0x1a8b0d0, Uty=<optimized out>, Uty@entry=0x7fffffffcc40, 
    W=0x1a57c30, y=0x7fffffffcc10, gwasnps=...) at src/lmm.cpp:1629
#6  0x000000000047fc44 in LMM::AnalyzePlink (this=this@entry=0x7fffffffcca0, U=U@entry=0x1a8b020, eval=eval@entry=0x1a8afb0, 
    UtW=UtW@entry=0x1a8b0d0, Uty=Uty@entry=0x7fffffffcc40, W=W@entry=0x1a57c30, y=0x7fffffffcc10, gwasnps=...)
    at /gnu/store/758qfjvn4wnsrhrr08pmwfr77vmcq4q2-gcc-8.2.0/include/c++/bits/stl_tree.h:129
#7  0x000000000045b214 in GEMMA::BatchRun (this=this@entry=0x7fffffffe1f0, cPar=...)
    at /gnu/store/758qfjvn4wnsrhrr08pmwfr77vmcq4q2-gcc-8.2.0/include/c++/bits/stl_set.h:223
#8  0x000000000048d6ad in main (argc=13, argv=0x7fffffffe388) at src/main.cpp:86
```

**Comment by pjotrp** [comment]
The bug is the batch compute which fails with a 0 size of the matrix!

**Comment by pjotrp** [comment]
Fixed with b4ec307c441d1e2a41da6f9665c79cc96b55da01
